src/components/select/select-item.ts

```diff
@@ -0,0 +1,48 @@
+import { queryAssignedNodes } from 'lit/decorators.js';
+import IgcDropdownItemComponent from '../dropdown/dropdown-item';
+
+/**
+ * Represents an item in a select list.
+ *
+ * @element igc-select-item
+ *
+ * @slot - Renders the all content bar the prefix and suffix.
+ * @slot prefix - Renders content before the main content area.
+ * @slot suffix - Renders content after the main content area.
+ *
+ * @csspart prefix - The prefix wrapper.
+ * @csspart content - The main content wrapper.
+ * @csspart suffix - The suffix wrapper.
+ */
+export default class IgcSelectItemComponent extends IgcDropdownItemComponent {
+  @queryAssignedNodes({ flatten: true })
+  private content!: Array<Element>;
+
+  public static override readonly tagName =
+    'igc-select-item' as 'igc-dropdown-item';
+
+  constructor() {
+    super();
+  }
+
+  /** Returns the text of the item without the prefix and suffix content. */
+  public override get textContent() {
+    return this.content
+      .map((t) => t.textContent?.trim())
+      .filter((t) => t !== '')
+      .join(' ');
+  }
+
+  /** Sets the textContent of the item without touching the prefix and suffix content. */
+  public override set textContent(value: string) {
```

Comment:
Except the prefix and suffix nodes are part of the textContent otherwise as well. This leads to them being included as input value, which is shown when an item is selected. This is bad UX IMO. Also, setting textContent on an `igc-select-item` will remove all prefix/suffix nodes otherwise. Overriding the textContent methods seemed like the best solution to me.